### ISSUE TITLE
job-tracking Finalizers prevent pod from deleting

### ISSUE THREAD

**Issue by josepperna** [kubernetes/kubernetes#114366]
### What happened?

When upgrading from kubernetes 1.22.11 to 1.23.9, some failed pods would get stuck on the Terminating state. We upgraded to 1.24.6 but these terminating pods did not disappear. Related to #109429 and #109486

Using `kubectl delete pod <pod_name> --force --grace-period=0` hangs and does not delete the pod. 

Using `kubectl patch pod <pod_name> -p '{"metadata":{"finalizers":null}}'` to remove the finalizers also does not work and returns the following error

```
The Pod <pod_name> is invalid: spec.tolerations: Forbidden: existing toleration can not be modified except its tolerationSeconds
```

### What did you expect to happen?

I expected the pod finalizers to be removed and the pod to be deleted.

### How can we reproduce it (as minimally and precisely as possible)?

Same steps as here #109429

### Kubernetes version

<details>

```console
$ kubectl version
1.24.6
```

</details>

### Cloud provider

<details>
AKS (Azure)
</details>

### OS version

<details>

```console
Ubuntu 18.04
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by kkkkun** [comment]
I have meeted this recently. It's caused by mutate-webhook which would add a toleration to Pod when creating, and then Pod would have two tolerations which have same name and different value.  

In kube-apiserver, I enable params `PodTolerationRestriction`  which would merge tolerations with name when call admission. 
The new object and  old object  would have different tolerations.  So validate-webhook refused the requests. 


I hope that my experience would give some advises.

**Comment by alculquicondor** [comment]
cc @mimowo

**Comment by alculquicondor** [comment]
Can you share your job yaml?

In particular, I wonder if you are using podFailurePolicy. If so, this is a known issue that is resolved in 1.27

> Using `kubectl patch pod <pod_name> -p '{"metadata":{"finalizers":null}}'` to remove the finalizers also does not work and returns the following error
> 
> ```
> The Pod <pod_name> is invalid: spec.tolerations: Forbidden: existing toleration can not be modified except its tolerationSeconds
> ```

It's possible that this failure is due to faulty webhook on your side that attempts to change the toleration. We have observed this before.
